db/migrate/20220304192308_create_rule_group_relationships.rb

```diff
@@ -0,0 +1,12 @@
+class CreateRuleGroupRelationships < ActiveRecord::Migration[7.0]
+  def change
+    create_table :rule_group_relationships, id: :uuid do |t|
+      t.references :left, type: :uuid, polymorphic: true
+      t.references :right, type: :uuid, polymorphic: true
+      t.string :relationship
+      t.index [:left_id, :right_id, :right_type, :left_type, :relationship],
```

Comment:
@vkrizan do you think we need this kind if complex index for this case? I have some doubts and maybe the validation should be good enough :thinking: